app/src/main/kotlin/com/simplemobiletools/notes/activities/MainActivity.kt

```diff
@@ -220,21 +246,37 @@ class MainActivity : SimpleActivity(), ViewPager.OnPageChangeListener {
 
     private fun openFile() {
         FilePickerDialog(this) {
-            val file = File(it)
-            if (file.isImageVideoGif()) {
-                toast(R.string.invalid_file_format)
-            } else if (file.length() > 10 * 1000 * 1000) {
-                toast(R.string.file_too_large)
-            } else if (mDb.doesTitleExist(it.getFilenameFromPath())) {
-                toast(R.string.title_taken)
-            } else {
-                OpenFileDialog(this, it) {
+            openFile(it, {
+                OpenFileDialog(this, it.path) {
                     addNewNote(it)
                 }
-            }
+            })
         }
     }
 
+    private fun openFile(path: String, onChecksPassed: (file: File) -> Unit, checkTitle: Boolean = true) {
```

Comment:
use anonymous functions as "onChecksPassed" as the last variable, lines 249+ are hard to read like this